src/prefect/agent/kubernetes/agent.py

```diff
@@ -836,7 +836,7 @@ def generate_deployment_yaml(
 
         version = prefect.__version__.split("+")
         image_version = (
-            "latest" if len(version) > 1 or latest else (version[0] + "-python3.6")
+            "latest" if len(version) > 1 or latest else (version[0] + "-python3.9")
```

Comment:
This is a relatively serious breaking change — we can bump to 3.8 in 1.0, but we can't just bump the default python version here.

```suggestion
            "latest" if len(version) > 1 or latest else (version[0] + "-python3.6")
```